### ISSUE TITLE
Schedule StartTimeOfDay sometimes returns UTC sometimes not

### ISSUE THREAD

**Issue by azturner** [SparkDevNetwork/Rock#5632]
### Description

There are several places that we are using the Schedule's StartTimeOfDay property in Lava to show a time. However when 
formatting this value with the Lava Date filter, it sometimes returns the actual time configured in the schedule, and sometime it returns a time adjusted for UTC. 

We first started noticing this in v14, but it doesn't seem to be tied to a specific version. We see the UTC value returned in one instance running v15.1 and not in another running v15.1 (both using Fluid engine). 

For example this Lava:

```
{% schedule where:'Id == "4"' %}
    <h4>{{ schedule.Name }} [ID:{{ schedule.Id }}] StartTimeOfDay</h4>
    {{ schedule.StartTimeOfDay }}<br/>
    {{ schedule.StartTimeOfDay | Date:'hh:mm tt' }} <small><i>(Formatted with 'hh:mm tt')</i></small><br/>
    <br/>
    <pre>{{ schedule.iCalendarContent }}</pre>
{% endschedule %}
```
Will return inconsistent results:
![image](https://github.com/SparkDevNetwork/Rock/assets/548180/7e1b5713-4e60-464b-ba18-096572ea5f49)

In those screen shots, the left instance is v15.1 and the right is v16, but we've seen a v15.1 Rock instance return the same as the v16 screenshot above. For example, this is a v15.1 instance:

![image](https://github.com/SparkDevNetwork/Rock/assets/548180/6b586b86-cc4f-4451-9061-16943443f5f7)

I'm not sure which of these is correct, but it would be helpful if the results were always consistent (either with UTC offset or not)

### Actual Behavior

Results are inconsistent

### Expected Behavior

Results to be consistent and predictable.

### Steps to Reproduce

Use the Lava above.

### Issue Confirmation

- [X] Perform a search on the [Github Issues](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ARock) to see if your bug or enhancement is already reported.
- [X] Try to reproduce the problem on a fresh install or on the [demo site](https://rock.rocksolidchurchdemo.com/).

### Rock Version

v14, v15, v16

### Client Culture Setting

en-US

**Comment by nairdo** [comment]
@azturner David, We've tracked this down to a difference between DotLiquid and Fluid.  Can confirm that the environments where you saw that situation were using different Lava engines?

![image](https://github.com/SparkDevNetwork/Rock/assets/383807/a515d77f-9492-475b-bac3-d96984dc338c)

Regardless, I don't believe that our Date filter (when using Fluid) should be converting the ...T103000 (10:30 AM) to 3:30 AM.  According to the iCal spec, a date-time value without the UTC designator or time zone are:

> said to be "floating" and are not bound to any time zone in particular. They are used to represent the same hour, minute, and second value regardless of which time zone is currently being observed.

So we'll look into that aspect.

**Comment by azturner** [comment]
@nairdo , thank you for looking into this. I had suspected it might be a Fluid vs DotLiquid, but I tried updating the site it was working correctly on to use Fluid (v15tm), and it didn't change anything. Although when I look at that site today, it is no longer working, so I'm guessing that maybe changing Lava engine requires restart. And yes, all of the sites where we've seen this issue are using Fluid.

**Comment by nairdo** [comment]
> so I'm guessing that maybe changing Lava engine requires restart.

Yes, that's right.  (_Unfortunately being a general global attribute, we really only have the tooltip to mention that:_)
![image](https://github.com/SparkDevNetwork/Rock/assets/383807/20a3e883-5986-4b75-b9c1-5b8578e13ca7)

**Comment by azturner** [comment]
@nairdo , is there any chance this fix can get included in v15.3?

**Comment by nairdo** [comment]
@azturner  We're in super crunch time with v15.3 about to go to alpha and v16.1 right behind that -- so I can't promise anything but we'll take a quick look to see how possible it would be to cherry-pick this back to v15.3.
